mantidproject/mantid
# Indirect - Data Analysis - Extract Convolution Fit Members

Currently, there is no option for extracting members (e.g. Data, Calc, Diff, Delta, Lorentzian) from the result of a convolution fit, into their own workspaces.

An 'ExtractMembers' boolean property has been added to the ConvolutionFitSequential algorithm, to provide users the option to extract members, each into their own workspace, which are subsequently grouped into a *_Members group workspace. The ConvFit interface has been updated to reflect this change, through the addition of a checkbox labelled 'ExtractMembers' in the properties table.

**To test:**
1. Navigate to Interfaces > Indirect > Data Analysis.
2. Click on the 'ConvFit' tab.
3. Supply a reduced (*_red.nxs) file into the Sample field (test files provided below).
4. Supply a resolution (*_res.nxs) file into the Resolution field (test files provided below).
5. Set Fit Type to One Lorentzian (for test files) or to the function whose shape best fits the miniplot.
6. In the properties table, check the checkbox with the label 'ExtractMembers'.
7. Run the algorithm.
8. Ensure a *_Members workspace was created.
11. Open a workspace from inside the *_Workspaces group workspace inside the viewer (table view).
12. Take note of the the names of all the members (row labels).
13. Expand the *_Members workspace and check that a workspace for every member is included:
      Name of member workspaces will have format *_NameOfMember
      **NOTE:** 'Convolution' member should be correctly renamed to 'Lorentzian1', 'Lorentzian2', 'Delta'                
                       depending on the function which was used in the convolution fitting, represented by that member.
14. Open the input sample (*_red) workspace and for each member workspace check that is has the 
      same number of spectrum (in this case - rows) as the sample workspace.
15. For each member workspace, check that it has the same X-values (Q-space) as the sample workspace, 
      excluding the first x-value in the sample workspace.

[ConvFitTestFiles.zip](https://github.com/mantidproject/mantid/files/1205356/ConvFitTestFiles.zip)



Fixes #20168. 

**Release Notes** 

#### Changes have been added to the release notes and workflow diagram has been updated.
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
